editor/src/messages/portfolio/document/node_graph/node_graph_message_handler/document_node_types.rs

```diff
@@ -2069,6 +2069,69 @@ fn static_nodes() -> Vec<DocumentNodeType> {
 			properties: node_properties::index_node_properties,
 			..Default::default()
 		},
+		// applies the given color to each pixel of an image
+		DocumentNodeType {
+			name: "Color Fill",
+			category: "Image Adjustments",
+			identifier: NodeImplementation::proto("graphene_core::raster::adjustments::ColorFillNode<_>"),
+			inputs: vec![
+				DocumentInputType::value("Image", TaggedValue::ImageFrame(ImageFrame::empty()), true),
+				DocumentInputType::value("Solid Color", TaggedValue::Color(Color::BLACK), false),
+			],
+			outputs: vec![DocumentOutputType::new("Image", FrontendGraphDataType::Raster)],
+			properties: node_properties::color_fill_node,
+			..Default::default()
+		},
+		DocumentNodeType {
+			name: "Color Overlay",
+			category: "Image Adjustments",
+			identifier: NodeImplementation::DocumentNode(NodeNetwork {
+				// this maps node inputs to the document nodes that will consume them
+				inputs: vec![0, 1, 2, 2],
+				// this maps to the outputs from the document nodes
+				outputs: vec![NodeOutput::new(2, 0)],
+				nodes: [
+					DocumentNode {
+						name: "Identity".to_string(),
+						inputs: vec![NodeInput::Network(concrete!(ImageFrame<Color>))],
+						implementation: DocumentNodeImplementation::Unresolved(NodeIdentifier::new("graphene_core::ops::IdNode")),
+						..Default::default()
+					},
+					DocumentNode {
+						name: "Color Fill".to_string(),
+						inputs: vec![NodeInput::node(0, 0), NodeInput::Network(concrete!(Color))],
+						implementation: DocumentNodeImplementation::Unresolved(NodeIdentifier::new("graphene_core::raster::adjustments::ColorFillNode<_>")),
+						..Default::default()
+					},
+					DocumentNode {
+						name: "Blend".to_string(),
+						inputs: vec![
+							NodeInput::node(1, 0),
+							NodeInput::node(0, 0),
+							NodeInput::Network(concrete!(BlendMode)),
+							NodeInput::Network(concrete!(f32)),
+						],
+						implementation: DocumentNodeImplementation::Unresolved(NodeIdentifier::new("graphene_core::raster::BlendNode<_, _, _, _>")),
+						..Default::default()
+					},
+				]
+				.into_iter()
+				.enumerate()
+				.map(|(id, node)| (id as NodeId, node))
+				.collect(),
+
+				..Default::default()
+			}),
+			inputs: vec![
+				DocumentInputType::value("Image", TaggedValue::ImageFrame(ImageFrame::empty()), true),
+				DocumentInputType::value("Color", TaggedValue::Color(Color::BLACK), true),
```

Comment:
Should this be exposed by default?

Most of the other nodes hide the optional parameters by default (by specifying false here).

Reply:
Yes you're right, I think the params should be hidden as well, since we are most likely going to receive the image from the connecting node, and the colour from the properties panel. 
I'll update the code to reflect this change.